common/app/common/NewNavigation.scala

```diff
@@ -498,19 +498,19 @@ object NewNavigation {
 
     def getSectionOrTagId(page: Page) = {
       val tags = Navigation.getTagsFromPage(page)
-      val commonKeywords = tagPages.intersect(tags.keywordIds)
+      val commonKeywords = tags.keywordIds.intersect(tagPages)
```

Comment:
As discussed, (being fussy) it would be nice if we could sort this explicitly and not rely on the implementation of `intersect`